Add initial grafana dashboard configuration

trailtacos opened the pull request:
In theory this support should work, but it is not 100% tested
as it has been extracted from my own custom implementation.

trailtacos committed:
Add initial grafana dashboard configuration

In theory this support should work, but it is not 100% tested
as it has been extracted from my own custom implementation.

michaelkamprath commented on grafana/grafana/dashboards/Provisioned/homebrew_aqi_with_env.json:
```diff
@@ -0,0 +1,1587 @@
+      "steppedLine": false,
+      "targets": [
+        {
+          "alias": "Indoor Temperature",
```
What if the sensor is not indoor?

trailtacos replied:
The dashboards are intended to be starting points :)  FYI, I believe these will average all the values from all the BME sensors - the entire dashboard is assuming they all live in the same environment, which obviously isn't the always the case.

I wonder if the sensors should be configured for whether they are inside or outside, or otherwise have a "location" configuration value that they expose through the JSON, that could then be used to differentiate them...

michaelkamprath replied:
So, in my case, I have 3 sensors going. I would prefer to keep them separate rather than averaging them together.

michaelkamprath commented on grafana/grafana/dashboards/Provisioned/homebrew_aqi_with_env.json:
```diff
@@ -0,0 +1,1587 @@
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Indoor temperature",
```
Again, what if the sensor is not indoors?

michaelkamprath commented on grafana/grafana/dashboards/Provisioned/homebrew_aqi_with_env.json:
```diff
@@ -0,0 +1,1587 @@
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Indoor humidity",
```
Again, what if the sensor is not indoors?

michaelkamprath commented on grafana/grafana/dashboards/Provisioned/homebrew_aqi_with_env.json:
```diff
@@ -0,0 +1,1587 @@
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Indoor Pressure",
```
Again, what if the sensor is not indoors?

michaelkamprath commented on grafana/README.md:
```diff
@@ -0,0 +1,81 @@
+Setup
+-----
+
+1. Edit `docker-compose.yaml` and change the IP address (192.168.0.2) to
```
Could we use an `.env` file (or one specified with `env_file`) instead of editing the docker compose file?

trailtacos replied:
I was thinking... I'm not even sure why that IP address is in the docker-compose - it shouldn't be needed - I'll play about with that (I took the original docker-compose from someone else and have altered and fixed it in other ways)

michaelkamprath commented on grafana/README.md:
```diff
@@ -0,0 +1,81 @@
+1. Edit `docker-compose.yaml` and change the IP address (192.168.0.2) to
+   your server IP address (the machine where the docker environment will run)
+
+2. Edit `telegraf/telegraf.d/inputs.homebrew_aqi.conf` to point at your
```
Could this be changed to used [an environment variable](https://github.com/influxdata/telegraf/blob/master/docs/CONFIGURATION.md#environment-variables) passthrough docker compose via a `.env` file?

trailtacos replied:
Interesting - I wonder if that works for lists... will try

michaelkamprath commented on grafana/telegraf/telegraf.d/inputs.homebrew_aqi.conf:
```diff
@@ -0,0 +1,26 @@
+  name_override = "homebrew_aqi"
+  interval = "10s"
+  urls = [
+    "http://192.168.0.11/json",
+    "http://192.168.0.47/json"
```
See my comment about maybe using environment variables above.

michaelkamprath commented on grafana/telegraf/telegraf.conf:
```diff
@@ -0,0 +1,14 @@
+  flush_interval = "10s"
+  flush_jitter = "0s"
+  precision = ""
+  hostname = "192.168.0.2"
```
Does this need to get edited by the used to change the IP address? You README doesn't say. Also, could it be controlled by an environment variable I set in the docker compose file?

trailtacos replied:
Oh urgh, good spot - I wonder what that is used for - I forgot about that one...

michaelkamprath reviewed (changes requested):
I got this working on my end, so in general this works. However, I generally dislike having to edit docker compose files and buried configuration files in a docker build. What are you thoughts on moving configurable things to an environment variable file?

Also, for the preconfigured dashboard with he environment sensors, the dashboard assumes there is only one sensor with the BME680. The AQI graph correct plots all separate sensor lines, but the environment graphs only plots one of the sensors (I think the first one configured).

trailtacos committed:
Move critical configuration to a `config.env` file

This means no docker-compose or telegraf configuration for
a basic setup.

trailtacos committed:
Small dashboard updates

1) Move them into an "Examples" directory
2) Ensure multiple lines for environment timelines

michaelkamprath commented on grafana/config.env:
```diff
@@ -0,0 +1,12 @@
+# This file contains basic configuration for telegraf
+
+# Should all metrics omit the host=$hostname tag
+OMIT_HOSTNAME=true
```
What does this mean? The comment does not give me understanding.

michaelkamprath reviewed (approved):
Works beautifully now with a nice concise single file to edit.
